### PR TITLE
DDF-2625: Fixes case insensitive enumeration validator

### PR DESCRIPTION
**Pull request by rzwiefel**
#### What does this PR do?
Fixes Enumeration Validator to not mutate the values passed in. 
#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged)?
@troymohl @bdeining @andrewkfiedler 

[Core APIs](https://github.com/orgs/codice/teams/core-apis)
#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).

@coyotesqrl
@jlcsmith

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2625

### CODE REVIEW

**Comment by troymohl** [comment]
My only input is that this doesn't obey the Set interface as this implementation is inconsistent with .equals. Since we never return the Set directly it should be ok. The ticket seems to indicate that  `AttributeValidationReport.suggestedValues` needs to match something (original json file?) for the UI to work. We should probably add that information to the javadoc for `AttributeValidationReport` in case other validators are added that don't meet that expectation.

**Comment by rzwiefel** [comment]
@troymohl It does obey the set interface, based upon your comparator. https://docs.oracle.com/javase/8/docs/api/java/util/TreeSet.html If your comparator is not, then it will be inconsistent. 

And also I would argue that you should **never** modify input data unless you explicitly are told it is allowed. You should always assume its not safe to mutate the input data and store it as is.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java [hunk 1]
```diff
@@ -54,10 +53,8 @@ public EnumerationValidator(final Set<String> values, final boolean ignoreCase)
 
         this.ignoreCase = ignoreCase;
 
-        this.values = Collections.unmodifiableSet(values.stream()
-                .filter(Objects::nonNull)
-                .map(ignoreCase ? String::toLowerCase : String::toString)
-                .collect(Collectors.toSet()));
+        this.values = new TreeSet<>(ignoreCase? String.CASE_INSENSITIVE_ORDER : null);
```
Comment (rzwiefel) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java]: Probably

Comment (rzwiefel) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java]: Though clearly checkstyle doesn't care. And in this case it makes it slightly more terse, so ima leave it in less someone has really strong opinions.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java [hunk 2]
```diff
@@ -54,10 +53,8 @@ public EnumerationValidator(final Set<String> values, final boolean ignoreCase)
 
         this.ignoreCase = ignoreCase;
 
-        this.values = Collections.unmodifiableSet(values.stream()
-                .filter(Objects::nonNull)
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java]: Are you no longer worried about there being a null value in the incoming set? You'll throw an NPE on the call to `addAll` below.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java [hunk 3]
```diff
@@ -54,10 +53,8 @@ public EnumerationValidator(final Set<String> values, final boolean ignoreCase)
...
-                .map(ignoreCase ? String::toLowerCase : String::toString)
-                .collect(Collectors.toSet()));
+        this.values = new TreeSet<>(ignoreCase? String.CASE_INSENSITIVE_ORDER : null);
+        this.values.addAll(values);
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java]: Could also change to something like...
```java
Supplier<Set<String>> supl =
        () -> ignoreCase ? new TreeSet<>(String.CASE_INSENSITIVE_ORDER) : new TreeSet();
this.values = values.stream()
        .filter(Objects::nonNull)
        .collect(Collectors.toCollection(supl));
```
...or take that and wrap in an unmodifiable collection is previously.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java (hunk 2 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java]: They cannot send a null value in the set.

Comment (coyotesqrl) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java]: There's nothing preventing it, other than your knowledge of the callers.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java (hunk 3 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java]: this is argubly is more complex.

and this set of values is never exposed anywhere, so there is no need to make it unmodifiable.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java (hunk 2 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java]: Filtering out nulls now.

File: catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java (hunk 1 above)
Comment (rzwiefel) [inline catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/validation/impl/validator/EnumerationValidator.java]: Looks like the formatter did it anyways for you.

**Pull request merged**